Sources/ScipioKit/Producer/FrameworkProducer.swift

```diff
@@ -74,94 +85,150 @@ struct FrameworkProducer {
             if isValidCache {
                 logger.info("✅ Valid \(product.target.name).xcframework is exists. Skip building.", metadata: .color(.green))
-                return
-            }
-            logger.warning("⚠️ Existing \(frameworkName) is outdated.", metadata: .color(.yellow))
-            logger.info("🗑️ Delete \(frameworkName)", metadata: .color(.red))
-            try fileSystem.removeFileTree(outputPath.absolutePath)
-            let restored = await cacheSystem.restoreCacheIfPossible(product: product)
-            needToBuild = !restored
-            if restored {
-                logger.info("✅ Restore \(frameworkName) from cache storage", metadata: .color(.green))
+                return true
+            } else {
+                let existsLocalFramework = fileSystem.exists(outputPath.absolutePath)
+                if existsLocalFramework {
+                    logger.warning("⚠️ Existing \(frameworkName) is outdated.", metadata: .color(.yellow))
+                    logger.info("🗑️ Delete \(frameworkName)", metadata: .color(.red))
+                    try fileSystem.removeFileTree(outputPath.absolutePath)
+                }
+                let restored = await cacheSystem.restoreCacheIfPossible(target: target)
+                if restored {
+                    logger.info("✅ Restore \(frameworkName) from cache storage", metadata: .color(.green))
+                    return true
+                } else {
+                    logger.warning("⚠️ Restoring \(frameworkName) is failed", metadata: .color(.yellow))
+                }
             }
-        } else {
-            needToBuild = true
         }
+        return false
+    }
 
-        if needToBuild {
-            switch product.target.type {
-            case .library:
-                let compiler = PIFCompiler(descriptionPackage: descriptionPackage, buildOptions: buildOptions)
-                try await compiler.createXCFramework(buildProduct: product,
-                                                     outputDirectory: outputDir,
-                                                     overwrite: overwrite)
-            case .binary:
-                guard let binaryTarget = product.target.underlyingTarget as? BinaryTarget else {
-                    fatalError("Unexpected failure")
-                }
-                let binaryExtractor = BinaryExtractor(
-                    package: descriptionPackage,
-                    outputDirectory: outputDir,
-                    fileSystem: fileSystem
-                )
-                try binaryExtractor.extract(of: binaryTarget, overwrite: overwrite)
-                logger.info("✅ Copy \(binaryTarget.c99name).xcframework", metadata: .color(.green))
-            default:
-                fatalError("Unexpected target type \(product.target.type)")
+    @discardableResult
+    private func buildXCFrameworks(
+        _ target: CacheSystem.CacheTarget,
+        outputDir: URL,
+        cacheSystem: CacheSystem
```

Comment:
Where is this cacheSystem used? I cannot find where `buildXCFrameworks` using it